src/Cardano/Wallet/Primitive/Types.hs

```diff
@@ -212,6 +220,39 @@ instance Buildable TxOut where
 instance Buildable (TxIn, TxOut) where
     build (txin, txout) = build txin <> " ==> " <> build txout
 
+data TxMeta = TxMeta
+    { metaId :: !(Hash "Tx")
+    -- FIXME: we might want to go for something more optimized than natural,
+    -- like Word64 ?
+    , depth :: !(Quantity "block" Natural)
+    , status :: !TxStatus
+    -- FIXME: should we use Quantity "lovelace" Natural ?
+    -- I wonder could we use somethine like proposed instead of Coin in wallet
+    -- part alltogether. Node/core bits would still work optimally with
+    -- Coin/Word64
+    -- but wallet bits could go a bit more abstract (and less optimized)?
+    , amount :: !Coin
```

Comment:
what about something less performant like `Quantity "lovelace" Natural` for representing coin in wallet?

EDIT: in fact ^ is probably a bad idea because exchanges are also our clients and we do want to have performant types in wallet

Reply:
I guess the `amount` is redundant with the transaction itself here. Not sure we do need to duplicate this information in the metadata.

Reply:
yes you are right - we can calculate this on demand. I thought this type will be use as a return type after calculation has been done already:
```haskell
computeTxMeta :: Tx -> ... -> TxMeta
```
so in `computeTxMeta` we would calculate this `amount`. As I am not sure how this type will be used yet through the code - I can remove it untill its needed